haskell-src/Concordium/Types.hs

```diff
@@ -1061,6 +1062,12 @@ newtype StateHashV0 = StateHashV0 {v0StateHash :: Hash.Hash}
 
 type StateHash = StateHashV0
 
+-- | Hash computed from the results in a block, such as block state, transaction outcomes and height
+-- information.
+-- This was first introduced as part of protocol version 7.
+newtype BlockResultHash = BlockResultHash {theBlockResultHash :: Hash.Hash}
+    deriving (Eq, Ord, Show, S.Serialize, ToJSON, FromJSON, FromJSONKey, ToJSONKey, Read, Hashable)
```

Comment:
You should probably use newtype deriving for this so that Show and Read instances just display the hash and not "BlockResultHash { the BlockResult... }".